src/OrbitQt/orbitmainwindow.cpp

```diff
@@ -1284,15 +1284,22 @@ void OrbitMainWindow::RestoreDefaultTabLayout() {
 
 void OrbitMainWindow::OnTimerSelectionChanged(const orbit_client_protos::TimerInfo* timer_info) {
   std::optional<int> selected_row(std::nullopt);
+  const uint64_t function_id = timer_info->function_id();
+  const auto live_functions_controller = ui->liveFunctions->GetLiveFunctionsController();
+  orbit_data_views::LiveFunctionsDataView* live_functions_data_view = nullptr;
+  if (live_functions_controller.has_value()) {
+    live_functions_data_view = &live_functions_controller.value()->GetDataView();
+  }
+  bool is_live_function_data_view_initialized = live_functions_data_view != nullptr;
   if (timer_info) {
-    uint64_t function_id = timer_info->function_id();
-    const auto live_functions_controller = ui->liveFunctions->GetLiveFunctionsController();
-    ORBIT_CHECK(live_functions_controller.has_value());
-    orbit_data_views::LiveFunctionsDataView& live_functions_data_view =
-        live_functions_controller.value()->GetDataView();
-    selected_row = live_functions_data_view.GetRowFromFunctionId(function_id);
-    live_functions_data_view.UpdateSelectedFunctionId();
-    live_functions_data_view.UpdateHistogramWithFunctionIds({function_id});
+    ORBIT_CHECK(is_live_function_data_view_initialized);
+    selected_row = live_functions_data_view->GetRowFromFunctionId(function_id);
+    live_functions_data_view->UpdateSelectedFunctionId();
+    live_functions_data_view->UpdateHistogramWithFunctionIds({function_id});
+  } else {
```

Comment:
Can you explain why the else case is not needed for the normal function selection case?

Reply:
What's "normal function selection"? Via `LiveFunctionsDataView::OnSelect` method?

Reply:
I mean where is the "diselection of the function happening". Here you are clearing the historgram.

Reply:
I think, the deselection is done by `ui->liveFunctions->OnRowSelected(selected_row);`. `selected_row.empty()` when `timer_info == nullptr`.

Reply:
I see. I was hoping that his code is less of spaghetti code. So apparently there is no central place to diselect/select a function...

Reply:
In general I find it a bit strange that you retrieve the function_id (from the timer info), and then call `UpdateSelectedFunctionId` which is setting the selected_function_id to the value in data manager (and not the your function_id). So this code assumes that the data manager has been already updated.

Reply:
OK, that's strange. Maybe @vickyliu-go4it or @karupayun can explain this.

Reply:
If `timer_info == nullptr`, then the `selected_row` is empty and `ui->liveFunctions->OnRowSelected(selected_row)` will clear the selection in the data view panel.

Reply:
Thanks for explaining this. Still it feels a bit strange. But it is a bit unrelated here.

Reply:
I think there is a hidden bug here. 

In the previous code, when we change the  timer selection in the capture window, we just changed the behavior in the data view panel ui, but we didn't store the selection in data view (to be specific, in `LiveFunctionsDataView::selected_indices`).

Reply:
Do you think this should be fixed in this PR?

Reply:
... and what's `LiveFunctionsDataView::selected_indices`?

Reply:
Ah, my bad, I fail to make myself clear again XD.

I mean it is fine if we just update the ui behavior and not store the selection in live data view. 

But as Florian pointed out, the orignal selection & update logic in the OrbitMainWindow::OnTimerSelectionChanged  is not very clear. Also live data view is the only data view whose will be affected by the timer selection in capture window. Live data view stores the selected function id, so it might make sense if we also update the stored selection in live data view when the selection from capture window changed.

But all in all, this is a tech debt exits long time ago before the histogram. So I will look into this after the presentation (or maybe in the fixit week)